tdms/tests/system/data/input_generation/matfile_option_edit.py

```diff
@@ -0,0 +1,99 @@
+import os
+from dataclasses import dataclass
+from pathlib import Path
+
+import numpy as np
+from hdf5storage import loadmat, savemat
+
+LOCATION_OF_THIS_FILE = os.path.abspath(os.path.dirname(__file__))
+
+
+@dataclass
+class OverwriteWith:
+    """Small container class that will be used when overwriting the "flag" variables usecd and intmethod when a config file specifies an 'adjust' field.
+
+    Intended workflow along the lines of:
+    intmethod = OverwriteWith(whether_to_overwrite, value_to_replace_with)
+    if intmethod.write:
+        h5data['intmethod'] = intmethod.value
+    else:
+        pass
+    """
+
+    # Flags whether or not to overwrite a value with self.value
+    write: bool
+    # The value to write if we are intending to replace a value in a file
+    value: bool | int | float | None
+
+
+def edit_mat_file(
+    test_directory: Path | str,
+    to_produce: Path | str,
+    options: dict[str, str | bool | int],
+) -> None:
+    """Produce a new .mat file by copying another .mat file and changing some of the TDMS "flag" variables according to the options provided.
+
+    This function can adjust the values of the following variables:
+    - usecd: Default value is 1 (FDTD). 0 results in use of PSTD
+    - intmethod: Default value is 1 (cubic). 2 results in BLI
+
+    :param test_directory: Directory in which the existing .mat file to use as a basis is stored.
+    :param to_produce: Filename to write the edited .mat file to.
+    :param options: Dictionary specifying the variables to adjust and the new values to assign to them.
+    """
```

Comment:
```suggestion
    """Create a new .mat file by copying an existing one and changing some TDMS "flag" variables according to options.
    
    The following `options` can be adjusted: 
    - usecd: Default is 1 (FDTD). Set to 0 for PSTD.
    - intmethod: Default is 1 (cubic). Set to 2 for BLI.

    :param test_directory: Directory containing the existing .mat file to use as a basis.
    :param to_produce: Filename for the edited .mat file.
    :param options (dict): The variable names to adjust and their new values.
    """
```
As usual, please check I didn't break the meaning.